### PR TITLE
move required, unwanted logic to api

### PR DESCRIPTION
**Pull request by rcerven**

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -1880,7 +1894,7 @@ def __init__(self, platform, release, platforms, arrangement_version,
                     .and_raise(CustomTestException))
 
         if raises_exception:
-            with pytest.raises(ValueError):
+            with pytest.raises(OsbsException):
```
Comment (vrutkovs) [inline tests/test_api.py]: The test should verify that underlying exception is `ValueError`

File: tests/test_api.py [hunk 2]
```diff
@@ -1812,10 +1823,10 @@ def test_reactor_config_secret(self):
         ("plat", 'rel', None, True, False, 1, False),
         ("plat", 'rel', None, True, False, None, True),
         # orchestrator build
-        (None, None, 'platforms', False, True, 1, True),
+        (None, None, 'platforms', False, True, 1, False),
```
Comment (vrutkovs) [inline tests/test_api.py]: Why wouldn't it raise exception here?

File: tests/test_api.py [hunk 3]
```diff
@@ -1812,10 +1823,10 @@ def test_reactor_config_secret(self):
...
         (None, None, 'platforms', False, True, None, False),
         # prod build
-        (None, None, None, False, False, 1, True),
+        (None, None, None, False, False, 1, False),
```
Comment (vrutkovs) [inline tests/test_api.py]: Why wouldn't it raise exception here?

File: tests/test_api.py (hunk 2 above)
Comment (rcerven) [inline tests/test_api.py]: arrangement_version is optional param for orchestrator, at least according docstring

- arrangement_version param may be used to select which                       
  orchestrator_inner:n.json template to use

File: tests/test_api.py (hunk 1 above)
Comment (rcerven) [inline tests/test_api.py]: added check

File: tests/fake_api.py [hunk 4]
```diff
@@ -364,6 +368,9 @@ def osbs(openshift, kwargs=None, additional_config=None):
             koji_root = http://koji.example.com/kojiroot
             koji_hub = http://koji.example.com/kojihub
             use_auth = false
+            git_branch = {git_branch}
```
Comment (lcarva) [inline tests/fake_api.py]: No, let's not set git information in config file.

File: tests/test_api.py [hunk 5]
```diff
@@ -1212,6 +1212,9 @@ def test_create_build_config_label_mismatch(self):
             has_ist_trigger=lambda: False,
             scratch=False)
 
+        spec = BuildSpec()
```
Comment (lcarva) [inline tests/test_api.py]: Why is this required now?

File: tests/test_api.py [hunk 6]
```diff
@@ -1840,18 +1851,21 @@ def __init__(self, platform, release, platforms, arrangement_version,
...
             'target': None,
             'architecture': None,
             'yum_repourls': None,
         }
-        if worker:
+        if arrangement_version:
             expected_kwargs.update({
                 'arrangement_version': arrangement_version,
+            })
+        if koji_upload_dir:
```
Comment (lcarva) [inline tests/test_api.py]: ??

File: tests/test_api.py (hunk 2 above)
Comment (lcarva) [inline tests/test_api.py]: I'm trying to understand what has changed that would change this behavior.

Comment (lcarva) [inline tests/test_api.py]: As @rcerven pointed out on IRC, this should not fail now because the CLI check incorrectly prohibited use of `arrangement_version` for orchestrator builds.

File: osbs/api.py [hunk 7]
```diff
@@ -652,6 +655,13 @@ def create_orchestrator_build(self, **kwargs):
         if not self.can_orchestrate():
             raise OsbsValidationException("can't create orchestrate build "
                                           "when can_orchestrate isn't enabled")
+        extra = set()
```
Comment (vrutkovs) [inline osbs/api.py]: `extra = [x for x in ('platform', 'release') if kwargs.get(x)]`?

File: tests/test_api.py (hunk 2 above)
Comment (vrutkovs) [inline tests/test_api.py]: Oh, I see, agreed

File: tests/test_api.py (hunk 3 above)
Comment (vrutkovs) [inline tests/test_api.py]: Same here - agreed

File: tests/test_api.py [hunk 8]
```diff
@@ -1345,7 +1345,7 @@ def test_create_build_config_create(self):
             .should_receive('create_build_config')
             .with_args(json.dumps(build_json))
             .once()
-            .and_return(flexmock(json=lambda: {'spam': 'maps'})))
+            .and_return(flexmock(json=lambda: {'spam': 'maps', 'spec': {}})))
```
Comment (vrutkovs) [inline tests/test_api.py]: Not sure why this change is required - nothing in current PR changes the output of `create_build_config`

File: tests/test_api.py [hunk 9]
```diff
@@ -1885,9 +1893,10 @@ def __init__(self, platform, release, platforms, arrangement_version,
                     .and_raise(CustomTestException))
 
         if raises_exception:
-            with pytest.raises(ValueError):
+            with pytest.raises(OsbsException):
                 cmd_build(MockArgs(platform, release, platforms, arrangement_version,
                           worker, orchestrator), osbs)
+                assert 'OsbsException: ValueError(' in caplog.text()
```
Comment (vrutkovs) [inline tests/test_api.py]: A better check would be:
```
with pytest.raises(OsbsException) as exc_info:
...
    assert isinstance(exc_info.cause, ValueError)
```

File: tests/test_api.py (hunk 8 above)
Comment (rcerven) [inline tests/test_api.py]: it doesn't anymore, will remove

File: tests/test_api.py (hunk 9 above)
Comment (rcerven) [inline tests/test_api.py]: sure

File: osbs/api.py (hunk 7 above)
Comment (rcerven) [inline osbs/api.py]: sure

File: tests/test_api.py (hunk 6 above)
Comment (rcerven) [inline tests/test_api.py]: on line 1829: koji_upload_dir = 'upload' if worker else None
koji_upload_dir is defined if worker

condition before was if worker, but that changed, so in order not to pass None koji_upload_dir, there is new condition for it

Comment (rcerven) [inline tests/test_api.py]: https://github.com/projectatomic/osbs-client/blob/master/osbs/cli/main.py#L384

koji_upload_dir is passed ONLY when is defined

Comment (vrutkovs) [inline tests/test_api.py]: It doesn't seem to be related to this PR, a separate commit/PR would be better

Comment (lcarva) [inline tests/test_api.py]: @rcerven, that makes sense. Otherwise, comparison of `expected_kwargs`, would fail because the `koji_upload_dir` key is set.

**Comment by lcarva** [comment]
Updated release note.
